app/views/contexts/section_summary_context.py

```diff
@@ -122,10 +122,25 @@ def _custom_summary_elements(self, section_summary, current_location, section):
 
     def _list_summary_element(self, summary, current_location, section) -> Mapping:
         current_list = self._list_store[summary["for_list"]]
-        list_collector_block = self._schema.get_list_collector_for_list(
-            section, for_list=summary["for_list"]
+
+        routing_path = self._router.routing_path(
```

Comment:
Now that we get the routing path here, we can change the method signature for the `_add_link` method so that it takes a routing path - it will save us accessing it twice

Reply:
Moved routing_path and current_location attributes out so add_link is a static method now. Thanks!

Reply:
I can see this was already being done (incorrectly) for the add link, but a routing path should not be being generated within a context; and doing so here will unnecessarily generate a routing path multiple times. Instead, the routing path should be passed to the context.

Reply:
I moved it out from context so it's being passed to it now. Thanks!